lib/runtime/wasmer/imports.go

```diff
@@ -325,62 +325,187 @@ func ext_misc_runtime_version_version_1(context unsafe.Pointer, z C.int64_t) C.i
+	if err != nil {
+		logger.Error("[ext_default_child_storage_exists_version_1] failed to get child from child storage", "error", err)
+		return 0
+	}
+	if child != nil {
+		return 1
+	}
 	return 0
 }
 
 //export ext_default_child_storage_get_version_1
-func ext_default_child_storage_get_version_1(context unsafe.Pointer, a, b C.int64_t) C.int64_t {
+func ext_default_child_storage_get_version_1(context unsafe.Pointer, childStorageKey, key C.int64_t) C.int64_t {
 	logger.Trace("[ext_default_child_storage_get_version_1] executing...")
-	logger.Warn("[ext_default_child_storage_get_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	storage := instanceContext.Data().(*runtime.Context).Storage
+
+	child, err := storage.GetChildStorage(asMemorySlice(instanceContext, childStorageKey), asMemorySlice(instanceContext, key))
+	if err != nil {
+		logger.Error("[ext_default_child_storage_get_version_1] failed to get child from child storage", "error", err)
+		return 0
+	}
+
+	value, err := toWasmMemoryOptional(instanceContext, child)
+	if err != nil {
+		logger.Error("[ext_default_child_storage_get_version_1] failed to allocate", "error", err)
+		panic(err)
+	}
+
+	return C.int64_t(value)
 }
 
 //export ext_default_child_storage_next_key_version_1
-func ext_default_child_storage_next_key_version_1(context unsafe.Pointer, a C.int64_t, b C.int64_t) C.int64_t {
+func ext_default_child_storage_next_key_version_1(context unsafe.Pointer, childStorageKey C.int64_t, key C.int64_t) C.int64_t {
 	logger.Trace("[ext_default_child_storage_next_key_version_1] executing...")
-	logger.Warn("[ext_default_child_storage_next_key_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	storage := instanceContext.Data().(*runtime.Context).Storage
+
+	child := storage.GetChildNextKey(asMemorySlice(instanceContext, childStorageKey), asMemorySlice(instanceContext, key))
+
+	value, err := toWasmMemoryOptional(instanceContext, child)
+	if err != nil {
+		logger.Error("[ext_default_child_storage_next_key_version_1] failed to allocate", "error", err)
+		panic(err)
+	}
+
+	return C.int64_t(value)
 }
 
 //export ext_default_child_storage_root_version_1
-func ext_default_child_storage_root_version_1(context unsafe.Pointer, z C.int64_t) C.int64_t {
+func ext_default_child_storage_root_version_1(context unsafe.Pointer, childStorageKey C.int64_t) C.int64_t {
 	logger.Trace("[ext_default_child_storage_root_version_1] executing...")
-	logger.Warn("[ext_default_child_storage_root_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	storage := instanceContext.Data().(*runtime.Context).Storage
+
+	// TODO: Should we write to DB or load from DB to state trie
+	// Commit() loads the state trie from DB
+	// WriteTrieToDB() writes the state trie to the DB
+	err := storage.Commit()
```

Comment:
you can remove this, the `Commit` is only used before the runtime execution actually starts

Reply:
In the spec, it is specially mentioned to commit. 
`Commits all existing operations and computes the resulting child storage root.`
So wouldn't this cause an issue?

Reply:
sorry, it's kind of a naming conflict, what the spec means is that all the changes have been put in the child trie, which has already happened at this point, so you can remove this